### PR TITLE
Rework of the competitive lobby

### PR DESCRIPTION
**Pull request by lostgeek**
Things left to do:

- [x] cancel / claim system
  - [x] Trigger this on disconnect
- [x] stopping a run at 3 losses
- [x] automatically share all replays of Angel Arena games 
- [x] show latest runs
- publishing runs (will do later)
- [x] in `eligible-players` check for previous opponents

Ideas
- open decklists

### CODE REVIEW

**Comment by NoahTheDuke** [comment]
This is looking good! Quick question: any particular reason to not hyphenate angel arena in the code?

**Comment by lostgeek** [comment]
Mostly because I didn't like to hyphenate the file name. So I decided to drop it for all code-related names. Should I add it back for consistency and rename the folder `angel_arena`?

**Comment by NoahTheDuke** [comment]
The style in the rest of the codebase is to hyphenate when joining two words. As file names are generally only referenced in require blocks, I don't mind if there is inconsistency between the variables/functions and the file name, but I think I'd prefer that we maintain consistency across the code itself.

File: src/clj/web/stats.clj [hunk 1]
```diff
@@ -174,8 +175,26 @@
                 :runner.agenda-points (get-in @state [:runner :agenda-point])}
      :history (:history @state)}))
 
+(defn handle-angelarena-stats
+  [db {:keys [ending-players original-players state format] :as game}]
+  (try
+    (doall
+      (for [player original-players]
```
Comment (NoahTheDuke) [inline src/clj/web/stats.clj:182]: If you're not using the return value of `for`, better to use `doseq` which has the same api but isn't lazy and returns `nil`.

File: src/clj/web/stats.clj [hunk 2]
```diff
@@ -174,8 +175,26 @@
                 :runner.agenda-points (get-in @state [:runner :agenda-point])}
      :history (:history @state)}))
 
+(defn handle-angelarena-stats
+  [db {:keys [ending-players original-players state format] :as game}]
+  (try
```
Comment (NoahTheDuke) [inline src/clj/web/stats.clj:180]: No catch block?

**Comment by lostgeek** [comment]
This is ready to be pushed live now! :tada:

File: src/clj/web/utils.clj [hunk 3]
```diff
@@ -30,3 +30,9 @@
        (.digest (MessageDigest/getInstance "MD5"))
        (BigInteger. 1)
        (format "%032x")))
+
+(defn average
+  [numbers]
+  (if (empty? numbers)
+    nil
```
Comment (NoahTheDuke) [inline src/clj/web/utils.clj:37]: Not 0?

File: src/css/angel-arena.styl [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+                    display-flex()
+                    width:100%!important
+                    height: 5rem
+                    margin: .25rem
+
+                    .info
+                        margin:auto
+                        flex-grow: 1
+
+
```
Comment (NoahTheDuke) [inline src/css/angel-arena.styl:185]: Lots of open space in this file. Can we clean it up a little?

**Pull request merged**